crypto/ube/ube_test.cc

```diff
@@ -47,7 +101,41 @@ TEST(Ube, MockedMethodTests) {
   ASSERT_TRUE(CRYPTO_get_ube_generation_number(&generation_number));
   ASSERT_EQ(generation_number, cached_generation_number);
 
-  // Mock another process fork. We used 10 before. Hence, 11 should work.
+  // Mock another snapsafe event but with a strictly smaller value.
+  set_snapsafe_generation_number_FOR_TESTING(1);
+
+  // Generation number should have incremented once.
+  cached_generation_number = generation_number;
+  generation_number = 0;
+  ASSERT_TRUE(CRYPTO_get_ube_generation_number(&generation_number));
+  ASSERT_EQ(generation_number, cached_generation_number + 1);
+
+  // Should be stable again.
+  cached_generation_number = generation_number;
+  generation_number = 0;
+  ASSERT_TRUE(CRYPTO_get_ube_generation_number(&generation_number));
+  ASSERT_EQ(generation_number, cached_generation_number);
+}
+
+TEST_F(ubeTest, MockedMethodTestsFork) {
```

Comment:
Checked test cases:
MockedMethodTestsSnapsafe:
* 5
* 6 increment by 1
* 8 incremenet by 2
* 1 decrement by a lot

MockedMethodTestsFork:
* 10
* 11 increment by 1
* 13 increment by 2
* 5 decrement by a lot

MockedMethodTestsForkAndSnapsafe:
* 20, 20
* 24, 23 both increment by a lot
* 1, 10 both decrement by a lot

Future possible values you could add:
1. Fork and snapsafe going in different directions
2. Rollover: max value -> 0
3. decrement by 1